astropy/time/core.py

```diff
@@ -662,19 +716,75 @@ def __getitem__(self, item):
         if self.isscalar:
             raise TypeError('scalar {0!r} object is not subscriptable.'.format(
                 self.__class__.__name__))
-        tm = self.replicate()
-        tm._time.jd1 = self._time.jd1[item]
-        tm._time.jd2 = self._time.jd2[item]
-        attrs = ('_delta_ut1_utc', '_delta_tdb_tt', 'location')
-        for attr in attrs:
-            val = getattr(self, attr, None)
-            if val is not None:
-                try:
-                    setattr(tm, attr, val[item])
-                except IndexError:  # location may be scalar (same for all)
-                    continue
+        return self._replicate('__getitem__', item)
 
-        return tm
+    def reshape(self, *args, **kwargs):
```

Comment:
Can you indicate for each method when the result is a view (memory shared) versus a copy?